static/app/views/alerts/rules/metric/ruleForm.tsx

```diff
@@ -1041,16 +1043,21 @@ class RuleFormContainer extends DeprecatedAsyncComponent<Props, State> {
     if (shouldShowOnDemandMetricAlertUI(this.props.organization)) {
       newState.isExtrapolatedChartData = Boolean(isExtrapolatedData);
     }
-    if (isLowConfidenceTimeSeries(data)) {
-      newState.isLowConfidenceChartData = true;
-    }
     this.setState(newState, () => this.fetchAnomalies());
     const {dataset, aggregate, query} = this.state;
     if (!isOnDemandMetricAlert(dataset, aggregate, query)) {
       this.handleMEPAlertDataset(data);
     }
   };
 
+  handleConfidenceTimeSeriesDataFetched(
+    data: EventsStats | MultiSeriesEventsStats | null
+  ) {
+    this.setState({isLowConfidenceChartData: isLowConfidenceTimeSeries(data)}, () =>
+      this.fetchAnomalies()
```

Comment:
I don't think we need to do this here since none of the user defined properties of the view are changing?